os/pm/pm_checkstate.c

```diff
@@ -145,6 +147,34 @@ enum pm_state_e pm_checkstate(int domain)
 		 */
 
 		(void)pm_update(domain, accum);
+#ifdef CONFIG_PM_METRICS
+		struct pm_accumchange_s *node = NULL;
+		if (pdom->state != pdom->recommended) {
+			struct pm_statechange_s *history = (struct pm_statechange_s*)dq_tail(&(pdom->history));
+			struct pm_accumchange_s *dest = NULL;
+
+			do {
```

Comment:
I think this seems strange. I know that after free(A), A will not be NULL, but the meaning looks strange.
How about like below?
```
dest = sq_peek();
while(dest) {
   free(dest);
   dest = sq_peek();
}
```